angular/projects/admin-nrpti/src/app/mines/mines-add-edit/mines-add-edit.component.ts

```diff
@@ -164,11 +184,56 @@ export class MinesAddEditComponent implements OnInit, OnDestroy {
    * @memberof MinesAddEditComponent
    */
   togglePublish(event) {
-    this.myForm.controls.publish.setValue(event.checked);
+    if (!event.checked) {
+      // always allow unpublishing
+      this.myForm.controls.publish.setValue(event.checked);
+    } else if (this.canPublish) {
+      // conditionally allow publishing
+      this.myForm.controls.publish.setValue(event.checked);
+    }
 
     this._changeDetectionRef.detectChanges();
   }
 
+  /**
+   * Return true if the record meets the criteria to be published, false otherwise.
+   *
+   * @returns {boolean}
+   * @memberof MinesAddEditComponent
+   */
+  checkCanPublish(): boolean {
+    return (
+      this.myForm.get('description').value &&
+      this.myForm.get('summary').value &&
+      this.myForm.get('type').value &&
+      this.mine.name &&
+      this.mine.status &&
+      this.mine.permitNumbers &&
+      this.mine.permitNumbers.length &&
+      this.mine.tailingsImpoundments &&
```

Comment:
I am not sure that all mines will have tailing impoundments. When they are imported it is possible they get set to '0' which would then fail this check.

Reply:
Yes, good catch.  Will update.